flink-runtime/src/test/java/org/apache/flink/runtime/io/network/buffer/BufferBuilderAndConsumerTest.java

```diff
@@ -104,8 +138,24 @@ private static ByteBuffer toByteBuffer(int... data) {
 		return byteBuffer;
 	}
 
-	private static void assertBufferContent(Buffer actualBuffer, int... expected) {
-		assertEquals(toByteBuffer(expected), actualBuffer.getNioBufferReadable());
+	private static void assertContent(BufferConsumer actualConsumer, int... expected) {
+		assertFalse(actualConsumer.isFinished());
+		Buffer buffer = actualConsumer.build();
+		assertFalse(buffer.isRecycled());
+		assertContent(buffer, expected);
+		assertEquals(expected.length * Integer.BYTES, buffer.getSize());
+		buffer.recycleBuffer();
+	}
+
+	private static void assertContent(Buffer actualBuffer, int... expected) {
+		IntBuffer actualIntBuffer = actualBuffer.getNioBufferReadable().order(ByteOrder.BIG_ENDIAN).asIntBuffer();
```

Comment:
shouldn't we make the same assumption on the byte order in `toByteBuffer()`?

Reply:
Dropped. It was some leftover code